// ng-modal/src/ng-modal.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from './ng-modal.component';
const styles_Modal:any[] = ([] as any[]);
export const RenderType_Modal:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Modal,
    data:{}});
function View_Modal_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'button',[['class',
      'close'],['data-dismiss','modal'],['type','button']],[[1,'aria-label',0]],[[(null as any),
      'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.close()) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,
      'span',[['aria-hidden','true']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['×']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.cancelButtonLabel || 'Close');
    _ck(_v,0,0,currVal_0);
  });
}
function View_Modal_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'h4',[['class',
      'modal-title']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.title;
    _ck(_v,1,0,currVal_0);
  });
}
function View_Modal_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'btn btn-default'],['data-dismiss','modal'],['type','button']],(null as any),
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.close()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.cancelButtonLabel;
        _ck(_v,1,0,currVal_0);
      });
}
function View_Modal_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'btn btn-primary'],['type','button']],(null as any),[[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.onSubmit.emit((undefined as any))) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.submitButtonLabel;
        _ck(_v,1,0,currVal_0);
      });
}
function View_Modal_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),28,'div',[['class',
      'modal-content'],['tabindex','0']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class','modal-header']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_Modal_2)),i0.ɵdid(16384,
          (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Modal_3)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),
      i0.ɵncd((null as any),0),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),3,'div',[['class','modal-body']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                '])),i0.ɵncd((null as any),1),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class','modal-footer']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                '])),i0.ɵncd((null as any),
          2),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_Modal_4)),i0.ɵdid(16384,
          (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Modal_5)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵted((null as any),['\n        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = !_co.hideCloseButton;
    _ck(_v,5,0,currVal_0);
    const currVal_1:any = _co.title;
    _ck(_v,8,0,currVal_1);
    const currVal_2:any = _co.cancelButtonLabel;
    _ck(_v,23,0,currVal_2);
    const currVal_3:any = _co.submitButtonLabel;
    _ck(_v,26,0,currVal_3);
  },(null as any));
}
export function View_Modal_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{modalRoot:0}),(_l()(),i0.ɵted((null as any),
      ['\n'])),(_l()(),i0.ɵeld(0,[[1,0],['modalRoot',1]],(null as any),11,'div',[['class',
      'modal'],['role','dialog'],['tabindex','-1']],(null as any),[[(null as any),
      'keydown.esc'],[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:i2.Modal = _v.component;
    if (('keydown.esc' === en)) {
      const pd_0:any = ((<any>(_co.closeOnEscape? _co.close(): 0)) !== false);
      ad = (pd_0 && ad);
    }
    if (('click' === en)) {
      const pd_1:any = ((<any>(_co.closeOnOutsideClick? _co.close(): 0)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({in:0,fade:1}),i0.ɵdid(278528,(null as any),
      0,i1.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},
      (null as any)),i0.ɵpod({display:0}),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),[[8,'className',
          0]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i2.Modal = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.preventClosing($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Modal_1)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),
          i0.ɵted((null as any),['\n'])),(_l()(),i0.ɵted((null as any),['\n']))],(_ck,
      _v) => {
    var _co:i2.Modal = _v.component;
    const currVal_0:any = 'modal';
    const currVal_1:any = _ck(_v,4,0,_co.isOpened,_co.isOpened);
    _ck(_v,3,0,currVal_0,currVal_1);
    const currVal_2:any = _ck(_v,6,0,(_co.isOpened? 'block': 'none'));
    _ck(_v,5,0,currVal_2);
    const currVal_4:any = _co.isOpened;
    _ck(_v,11,0,currVal_4);
  },(_ck,_v) => {
    var _co:i2.Modal = _v.component;
    const currVal_3:any = ('modal-dialog ' + _co.modalClass);
    _ck(_v,8,0,currVal_3);
  });
}
export function View_Modal_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'modal',([] as any[]),
      (null as any),(null as any),(null as any),View_Modal_0,RenderType_Modal)),i0.ɵdid(180224,
      (null as any),0,i2.Modal,([] as any[]),(null as any),(null as any))],(null as any),
      (null as any));
}
export const ModalNgFactory:i0.ComponentFactory<i2.Modal> = i0.ɵccf('modal',i2.Modal,
    View_Modal_Host_0,{modalClass:'modalClass',closeOnEscape:'closeOnEscape',closeOnOutsideClick:'closeOnOutsideClick',
        title:'title',hideCloseButton:'hideCloseButton',cancelButtonLabel:'cancelButtonLabel',
        submitButtonLabel:'submitButtonLabel'},{onOpen:'onOpen',onClose:'onClose',
        onSubmit:'onSubmit'},['modal-header','modal-content','modal-footer']);
const styles_ModalContent:any[] = ([] as any[]);
export const RenderType_ModalContent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_ModalContent,
    data:{}});
export function View_ModalContent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0)],(null as any),(null as any));
}
export function View_ModalContent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'modal-content',
      ([] as any[]),(null as any),(null as any),(null as any),View_ModalContent_0,
      RenderType_ModalContent)),i0.ɵdid(49152,(null as any),0,i2.ModalContent,([] as any[]),
      (null as any),(null as any))],(null as any),(null as any));
}
export const ModalContentNgFactory:i0.ComponentFactory<i2.ModalContent> = i0.ɵccf('modal-content',
    i2.ModalContent,View_ModalContent_Host_0,{},{},['*']);
const styles_ModalFooter:any[] = ([] as any[]);
export const RenderType_ModalFooter:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_ModalFooter,
    data:{}});
export function View_ModalFooter_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0)],(null as any),(null as any));
}
export function View_ModalFooter_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'modal-footer',
      ([] as any[]),(null as any),(null as any),(null as any),View_ModalFooter_0,RenderType_ModalFooter)),
      i0.ɵdid(49152,(null as any),0,i2.ModalFooter,([] as any[]),(null as any),(null as any))],
      (null as any),(null as any));
}
export const ModalFooterNgFactory:i0.ComponentFactory<i2.ModalFooter> = i0.ɵccf('modal-footer',
    i2.ModalFooter,View_ModalFooter_Host_0,{},{},['*']);
const styles_ModalHeader:any[] = ([] as any[]);
export const RenderType_ModalHeader:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_ModalHeader,
    data:{}});
export function View_ModalHeader_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0)],(null as any),(null as any));
}
export function View_ModalHeader_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'modal-header',
      ([] as any[]),(null as any),(null as any),(null as any),View_ModalHeader_0,RenderType_ModalHeader)),
      i0.ɵdid(49152,(null as any),0,i2.ModalHeader,([] as any[]),(null as any),(null as any))],
      (null as any),(null as any));
}
export const ModalHeaderNgFactory:i0.ComponentFactory<i2.ModalHeader> = i0.ɵccf('modal-header',
    i2.ModalHeader,View_ModalHeader_Host_0,{},{},['*']);
